pkg/queue/health/probe.go

```diff
@@ -144,18 +143,16 @@ func http2UpgradeProbe(config HTTPProbeConfigOptions) (int, error) {
 }
 
 // HTTPProbe checks that HTTP connection can be established to the address.
-func HTTPProbe(ctx context.Context, config HTTPProbeConfigOptions) error {
-	cfg := apicfg.FromContextOrDefaults(ctx)
-	autoDetect := cfg.Features.AutoDetectHTTP2
-	if autoDetect == apicfg.Disabled {
+func HTTPProbe(config HTTPProbeConfigOptions) error {
+	if !config.AutoDetectHTTP2 {
 		config.MaxProtoMajor = 1
 	}
 	// If we don't know if the connection supports HTTP2, we will try it.
 	// Once we get a non-error response, we won't try again.
 	// If maxProto is 0, container is not ready, so we don't know whether http2 is supported.
 	// If maxProto is 1, we know we're ready, but we also can't upgrade, so just return.
 	// If maxProto is 2, we know we can upgrade to http2
-	if autoDetect == apicfg.Enabled && config.MaxProtoMajor == 0 {
+	if config.AutoDetectHTTP2 && config.MaxProtoMajor == 0 {
```

Comment:
I feel like Im missing something here. If config.AutoDetectHTTP2 is false then config.MaxProtoMajor is always 1 due to the line above, so config.MaxProtoMajor=0 implies AutoDetectHTTP2 can only be true -> I think the first part of this does nothing, which makes me worry there was something important in the "enabled" vs "disabled" distinction before.. but I dont see what :(. Maybe it was just a mistake, or I need more coffee.

Reply:
I see your point, though I think I'd prefer keeping this for the sake of only changing what I intended to change in this PR.

Reply:
I guess my worry is _either_ we're confident enough that we understand whats going on to remove this, or we're not. We actually already are changing this in the PR, because before the two conditions werent the same (one compared autoDetect against the string "Disabled", one against "Enabled")

Reply:
Ah, so feature flags in theory have a third state named "Allowed" that in theory gives explicit access to the feature via annotations and thelike. That comparison would have to happen on the reconciler side then anyway, since that would then decide if this specific proxy enables the feature or not.

As such the comparison here **should** be just redundant and maybe serves as an explicit reminder that this path is enabled? Maybe we should drop the MaxProtoMajor comparison instead :thinking:

Reply:
yah from the configmap comment it seems like we only support enabled/disabled for this so thats good, just this logic makes me nervous we might have been doing something funky and I want to be sure we werent (obviously it could also have just been a mistake thats just more obvious now its a bool 🤷)

Reply:
Reading through the code again and I think what is supposed to be happening here, is that the MaxProtoMajor value is actually stored in between invocations of `HTTPProbe`, so that even if we're running multiple probes, we only probe HTTP2 once.

So what we can actually do is drop the setting from config and only key off of MaxProtoMajor being 0.